docs/specification/draft/basic/security_best_practices.mdx

```diff
@@ -139,3 +139,86 @@ Token passthrough is explicitly forbidden in the [authorization specification](/
+```
+
+#### 2.3.2 Session Highjack Impersonation
+
+```mermaid
+sequenceDiagram
+    participant Client
+    participant Server
+    participant Attacker
+
+    Client->>Server: Initialize (login/authenticate)
+    Server-->>Client: Respond with session ID (persistent session created)
+
+    Attacker->>Server: Access/guess session ID
+    Note right of Attacker: Attacker knows/guesses session ID
+
+    Attacker->>Server: Make API call (using session ID, no re-auth)
+    Server-->>Attacker: Respond as if Attacker is Client (session hijack)
+```
+
+#### 2.3.3 Attack Description
+
+When you have multiple stateful HTTP servers that handle MCP requests, the following attack vectors are possible:
+
+1. Session Hijack Prompt Injection
+  1. The client connects to **Server A** and receives a session ID.
+  2. The attacker obtains an existing session ID and sends a malicious event to **Server B** with said session ID.
+    - When a server supports [redelivery/resumable streams](https://modelcontextprotocol.io/specification/2025-03-26/basic/transports#resumability-and-redelivery), deliberately terminating the request before receiving the response could lead to it being resumed by the original client via the GET request for server sent events.
+    - If a particular server initiates server sent events as a consequence of a tool call such as a `notifications/tools/list_changed`, where it is possible to affect the tools that are offered by the server, a client could end up with tools that they were not aware were enabled.
+  3. **Server B** enqueues the event (associated with session ID) into a shared queue.
+  4. **Server A** polls the queue for events using the session ID and retrieves the malicious payload.
+  5. **Server A** sends the malicious payload to the client as an asynchronous or resumed response.
+  6. The client receives and acts on the malicious payload, leading to potential compromise.
+2. Session Impersonation Hijack
+  1. The MCP client authenticates with the MCP server, creating a persistent session ID.
+  2. The attacker obtains the session ID.
+  3. The attacker makes calls to the MCP server using the session ID.
+ 4. MCP server does not check for additional authorization and treats the attacker as a legitimate user, allowing unauthorized access or actions.
+
+#### 2.3.4 Mitigation
+
+To prevent session hijacking and event injection attacks, the following mitigations should be implemented:
+
+MCP servers that implement authorization **MUST** verify all inbound requests.
+MCP Servers **MUST NOT** use sessions for authentication.
+
+MCP servers **MUST** use secure, non-deterministic session IDs.
+Generated session IDs (e.g., UUIDs) **SHOULD** use secure random generators. Avoid predictable or sequential session identifiers that could be guessed by an attacker. Rotating or expiring session IDs can also reduce the risk.
+
+MCP servers **SHOULD** bind session keys to user-specific information.
```

Comment:
Do you mean to refer to "session IDs" rather than "keys" here? Or are session keys something else?